### DISCUSSION TITLE
What do these graphs look like to you?

### DISCUSSION THREAD

**Discussion by aspencoyle** [RobertsLab/resources#1559] [Q&A]
Alright, so some background info: this is related to my work using GLMMs to understand factors behind the spread of Hematodinium in SE AK Tanners. The model uses the following variables:

- Crab size
- Sex
- Latitude
- Depth
- Day of the year
- Injuries to crab
- Water temperature
- Time since last molt (using shell condition, a proxy)
- Presence of Black Mat (a fungal parasite)

Year and Site are also included as random effects (basically, my model treats each site as having a different baseline infection rate, along with each year).

When I run that model, water temperature comes out as highly significant (p < 2x10^-16), with more infected crabs at higher water temperatures. However, when I do a basic plot of water temperature vs. infection rate (no model whatsoever, just a basic scatterplot), the correlation flips. 

I dug into the data a bit more, and it looks like different sites have baseline differences in infection rates! I graphed each individual site's infection rate - all are below. Some important notes:

- Each point on the plot is the infection rate at a given temperature (0.1C intervals). 
- Size of the point corresponds to number of crabs at that point
- Axes are identical in all plots (temperature from 3C to 10C)
- Point size is NOT identical in all plots

Anyway, here are the graphs - I'll post my interpretation as a comment below, but don't want to bias you - take a peek at the graphs first!

![Barlow_Cove](https://user-images.githubusercontent.com/71358918/207467326-080583dc-3ede-4d40-aad9-14e67ffcca0a.png)
![Deadman_Reach](https://user-images.githubusercontent.com/71358918/207467330-f46ac2dc-129b-4028-809f-dea68239fa35.png)
![Excursion_Inlet](https://user-images.githubusercontent.com/71358918/207467331-07eb5143-9eab-4242-b25c-fc971858f651.png)
![Gambier_Bay](https://user-images.githubusercontent.com/71358918/207467334-a8191485-8c9f-4c87-a78a-36d9ce927f6b.png)
![Glacier_Bay](https://user-images.githubusercontent.com/71358918/207467335-b934a0b5-434b-4842-be51-46745daade3d.png)
![Holkham_Bay](https://user-images.githubusercontent.com/71358918/207467336-3ce3d2a9-dd88-4aff-aa1c-f385b8fd7ff7.png)
![Icy_Strait](https://user-images.githubusercontent.com/71358918/207467340-49706c39-f93a-4b1d-af98-523ec8325c1b.png)
![Juneau](https://user-images.githubusercontent.com/71358918/207467342-78aba9c6-4c31-443a-b57e-3b5032330d13.png)
![Lynn_Sisters](https://user-images.githubusercontent.com/71358918/207467343-043d800a-43e2-4dbb-981e-721bea0c3dbc.png)
![Port_Camden](https://user-images.githubusercontent.com/71358918/207467345-53a3a444-4bf2-4508-b48b-294a93dd9012.png)
![Port_Frederick](https://user-images.githubusercontent.com/71358918/207467347-775212d1-94e9-45e4-83da-3da631bb3122.png)
![Pybus_Bay](https://user-images.githubusercontent.com/71358918/207467314-dd292dc0-ead6-41ab-8389-1af73137880a.png)
![Rodman_Bay](https://user-images.githubusercontent.com/71358918/207467318-f2626368-acfa-474b-89e7-56997cf2f07d.png)
![Seymour_Canal](https://user-images.githubusercontent.com/71358918/207467319-dfc27136-2550-457d-a8bc-dccb58cb8807.png)
![St _James Bay](https://user-images.githubusercontent.com/71358918/207467321-bc828e5a-b0e5-4694-a702-d425da9ba1fa.png)
![Stephens_Passage](https://user-images.githubusercontent.com/71358918/207467323-f14f24de-7dbc-4037-8d78-76001df78517.png)
![Thomas_Bay](https://user-images.githubusercontent.com/71358918/207467324-05145eb3-38e8-44ad-b23c-3531c4a2d10e.png)

**Comment by aspencoyle** [comment]
My read is that the graphs fall into four basic categories - Positive Trend, Negative Trend, No Trend, and Minimal Infections. The first three are pretty self-explanatory, the fourth is basically that infections are really rare in these locations (typically well below 5%), which means we just have much less data to produce a trend line from. After all, what can you really say if you have 1,000 crabs of data but only 10 are infected?

Anyway, I sort them as following:

**Positive Trend**
- Barlow Cove
- Deadman Reach
- Juneau
- Lynn Sisters
- Rodman Bay (maybe minimal infections, little data from here)
- St. James Bay
- Thomas Bay

**Negative Trend**
- Excursion Inlet

**No Trend**
- Gambier Bay
- Holkham Bay
- Port Frederick (maybe positive?)
- Stephens Passage

**Minimal Infections**
- Glacier Bay
- Icy Strait
- Port Camden
- Pybus Bay
- Seymour Canal

So in total, that's 6-8 with a positive relationship to temperature, 1 with a negative trend, 3-4 with no trend, and 5-6 without sufficient data. To me, that doesn't seem to invalidate the model results (which showed that strong positive correlation with temperature). But what do you think? Post those comments below, and don't forget to like and subscribe!

**Comment by sr320** [comment]
Can you plot all the data (by location) and or provide link to data?

**Reply by aspencoyle** [reply]
Do you mean a plot of all the data together? Sure thing, here's that! Each site is a different color, with points at 1 degree intervals. Didn't link point size to sample size (gets too busy)

![bitter_with_site](https://user-images.githubusercontent.com/71358918/207710538-e182c9d0-fa89-48eb-959b-1329e0a70f25.png)
Also link to data: 

https://github.com/afcoyle/hemat_modeling/blob/main/output/ADFG_SE_AK_pot_surveys/cleaned_data/crab_data/BCS_examined_crab_with_temperature.csv

**Reply by AHuffmyer** [reply]
One way to plot this data would be to plot data by site as you have done in the graph above, but do a separate plot (facet_grid or facet_wrap) for each grouping of sites (e.g., one plot for no trend, one for positive, one for negative). Then adding sample size with point size wouldn't be too busy.

**Reply by sr320** [reply]
>https://github.com/afcoyle/hemat_modeling/blob/main/output/ADFG_SE_AK_pot_surveys/cleaned_data/crab_data/BCS_examined_crab_with_temperature.csv 

Link does not work

**Reply by sr320** [reply]
Agree with @AHuffmyer suggestion.. but also a version where data points are not aggregated.

**Reply by aspencoyle** [reply]
Ahh sorry, realized it was a private repo! Link should work now. Alternatively, [link to raw data](https://raw.githubusercontent.com/afcoyle/hemat_modeling/main/output/ADFG_SE_AK_pot_surveys/cleaned_data/crab_data/BCS_examined_crab_with_temperature.csv?token=GHSAT0AAAAAAB3SCLEBWA7L5FLVQN3W5PHCY42HOVQ)

And ooh sounds good, will work on creating that graph!

**Comment by sr320** [comment]
To me the most challenging aspect is data for bitter is binary (0 or 1) and you are averaging them.

**Reply by sr320** [reply]
What I would like to see if I understand the data is the change in proportion of positive along side change in temperature.

**Reply by aspencoyle** [reply]
Yep, that's definitely been an issue with visualizations - using infection rate seems to work alright, but wonder if there might be better alternatives..

**Reply by sr320** [reply]
Are you using infection rate (proportion infected) or average of bitter value?

**Reply by AHuffmyer** [reply]
You could plot binary data (binomial) with a regression line to view the relationship between infection and temperature, it might look something like this (image just from googling):  
![image](https://user-images.githubusercontent.com/32178010/207719785-48892d6a-fad0-4fae-80de-3d3ed68b6ab5.png)

**Reply by AHuffmyer** [reply]
Simple tutorial as an example here: https://www.statology.org/plot-logistic-regression-in-r/

**Reply by aspencoyle** [reply]
Steven - using average of bitter value, but that's the same as infection rate (infected = 1, uninfected = 0). 

Ariana - I considered that but figured it'd be ineffective with the amount of data - wouldn't hurt to give it a shot though! Thanks for the tutorial, should make it quick to actually plot!

**Reply by aspencoyle** [reply]
Also Steven, just saw this earlier comment from you: `What I would like to see if I understand the data is the change in proportion of positive along side change in temperature.`

Basically, but without the change aspect - x-axis is ambient bottom temperature at time of capture, y-axis is total proportion of visually positive crabs.

**Reply by sr320** [reply]
> Basically, but without the change aspect - x-axis is ambient bottom temperature at time of capture, y-axis is total proportion of visually positive crabs. 

Can you plot infection rate at each year per site along side temperature?

**Reply by aspencoyle** [reply]
Here's infection rate at each year per site along temperature! Don't want to clog up the readability of this discussion (and 17 sites is too many for facet_wrap to display effectively), so here's a link to them all on GitHub!

https://github.com/afcoyle/hemat_modeling/tree/main/output/ADFG_SE_AK_pot_surveys/diagnostic_images/site_specific_graphs/infection_rates/with_year

**Comment by aspencoyle** [comment]
Working on some other graphs, but here's all logistic regression graphs! They get a bit muddled due to the large numbers of data points, but here they are! 

Edit: they clog up a lot of space, so here's [the link](https://github.com/afcoyle/hemat_modeling/tree/main/output/ADFG_SE_AK_pot_surveys/diagnostic_images/site_specific_graphs/logistic_regression)

Here's a jittered and facet-wrapped version of the logistic regression graphs:
![all_sites](https://user-images.githubusercontent.com/71358918/207728748-d300516d-0a4a-471f-9335-bde1115cf08a.png)

**Reply by sr320** [reply]
_facet_wrap_ these and add some _jitter_.

**Reply by aspencoyle** [reply]
Edited and posted above!

**Comment by laurahspencer** [comment]
Scanning the plots- is it possible that there is a non-linear effect of temperature on infection rate?  For some of the locations I see possibly an "A" shaped pattern- where infection rate increases with temp, peaks, then decreases at higher temperatures.  Additionally, it looks like infection rate becomes more variable at higher temperatures.  Perhaps a simple correlation/scatter plot isn't able to pick up nonlinear trends.
